debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/SourceInfo.java

```diff
@@ -299,6 +300,20 @@ public void initialPosition(String replicaSetName, BsonDocument oplogEvent) {
         onEvent(replicaSetName, CollectionId.parse(replicaSetName, namespace), position, 0L);
     }
 
+    public void cursorEvent(String replicaSetName, MongoChangeStreamCursor<?> cursor) {
+        if (cursor == null || cursor.getResumeToken() == null) {
+            return;
+        }
+
+        String namespace = "";
+        long wallTime = 0L;
+        BsonTimestamp ts = ResumeTokens.getTimestamp(cursor.getResumeToken());
+        Position position = Position.changeStreamPosition(ts, cursor.getResumeToken().getString("_data").getValue(), null);
+        positionsByReplicaSetName.put(replicaSetName, position);
+
+        onEvent(replicaSetName, CollectionId.parse(replicaSetName, namespace), position, wallTime);
```

Comment:
⚠️ I could really use a second pair of eyes here to help verify the impact of setting some of these fields thusly. Had a hard time understanding all the implications of the various state elements, their invariants, etc.